### PR TITLE
Feature/call hanging

### PR DESCRIPTION
**Pull request by yotamss**

### CODE REVIEW

File: Pulse.Core/AudioStreaming/CallHanger.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            this.actualChannelReader = actualChannelReader;
+            this.callHanger = callHanger;
+            connectionLost = false;
+
+            lastPacketReceivedAt = DateTimeOffset.Now;
+            _ = HangIfNoResponseForTooLongAsync();
+        }
+
+        private async Task HangIfNoResponseForTooLongAsync()
```
Comment (GurGaller) [inline Pulse.Core/AudioStreaming/CallHanger.cs]: I would separate this from the 0-based hangup. They are not related to one another, even though their side effect is the same

Comment (GurGaller) [inline Pulse.Core/AudioStreaming/CallHanger.cs]:
```suggestion
        private async Task HangUpIfNoResponseForTooLongAsync()
```

File: Pulse.Core/AudioStreaming/CallHanger.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            while (true)
+            {
+                if (DateTimeOffset.Now - lastPacketReceivedAt > TimeSpan.FromSeconds(5))
+                {
+                    Console.WriteLine("Assuming connection lost, hanging up (without notifying the other side).");
```
Comment (GurGaller) [inline Pulse.Core/AudioStreaming/CallHanger.cs]: But you are notifying the other side...

**Review by GurGaller — CHANGES_REQUESTED** [review]

File: Pulse.Core/AudioStreaming/CallHanger.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            this.callHanger = callHanger;
+            connectionLost = false;
+
+            lastPacketReceivedAt = DateTimeOffset.Now;
+            _ = HangIfNoResponseForTooLongAsync();
+        }
+
+        private async Task HangIfNoResponseForTooLongAsync()
+        {
+            while (true)
```
Comment (GurGaller) [inline Pulse.Core/AudioStreaming/CallHanger.cs]: You need to stop this loop when the object is disposed

File: Pulse.Core/AudioStreaming/CallHanger.cs (hunk 1 above)
Comment (yotamss) [reply Pulse.Core/AudioStreaming/CallHanger.cs]: later

**Pull request merged**